src/OpenTelemetry.Api/Context/Propagation/AWSXRayPropagator.cs

```diff
@@ -0,0 +1,276 @@
+        private const char TraceIdDelimiter = '-';
+
+        private const string ParentKey = "Parent";
+        private const int ParentIdHexDigits = 16;
+
+        private const string SampledKey = "Sampled";
+        private const char SampledValue = '1';
+        private const char NotSampledValue = '0';
+
+        private static readonly char[] ValidSeparators = { ';' };
+        private static readonly HashSet<string> AllFields = new HashSet<string>() { AWSXRayTraceHeaderKey };
+
+        /// <inheritdoc/>
+        public ISet<string> Fields => AllFields;
+
+        /// <inheritdoc/>
+        public PropagationContext Extract<T>(PropagationContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+        {
+            if (context.ActivityContext.IsValid())
+            {
+                return context;
+            }
+
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToExtractActivityContext(nameof(AWSXRayPropagator), "null carrier");
+                return context;
+            }
+
+            if (getter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToExtractActivityContext(nameof(AWSXRayPropagator), "null getter");
+                return context;
+            }
+
+            try
+            {
+                var parentTraceHeader = getter(carrier, AWSXRayTraceHeaderKey);
+
+                if (parentTraceHeader == null || parentTraceHeader.Count() != 1)
+                {
+                    return context;
+                }
+
+                var parentHeader = parentTraceHeader.First();
+
+                if (!TryParse(parentHeader, out var rootTraceId, out var parentId, out var sampleDecision))
+                {
+                    return context;
+                }
+
+                var rootId = ExtractRootTraceId(rootTraceId);
+                var traceId = ActivityTraceId.CreateFromString(rootId.AsSpan());
+                var spanId = ActivitySpanId.CreateFromString(parentId.AsSpan());
+                var traceOptions = sampleDecision == SampledValue ? ActivityTraceFlags.Recorded : ActivityTraceFlags.None;
+
+                return new PropagationContext(new ActivityContext(traceId, spanId, traceOptions, isRemote: true), context.Baggage);
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetryApiEventSource.Log.ActivityContextExtractException(nameof(AWSXRayPropagator), ex);
+            }
+
+            return context;
+        }
+
+        /// <inheritdoc/>
+        public void Inject<T>(PropagationContext context, T carrier, Action<T, string, string> setter)
+        {
+            if (context.ActivityContext.TraceId == default || context.ActivityContext.SpanId == default)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "Invalid context");
```

Comment:
Do the other propagators log for this? I think inject into unsampled activity is no-op by design not an error

Reply:
Yes, just aligning with [B3Propagator](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs#L109) and [TextMapPropagator](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Api/Context/Propagation/TextMapPropagator.cs#L109)